api/timeseries.go

```diff
@@ -6,42 +6,47 @@ import (
 	"strings"
 	"text/template"
 
-	"github.com/ONSdigital/dp-search-query/elasticsearch"
 	"github.com/ONSdigital/go-ns/log"
+	"github.com/gorilla/mux"
 )
 
+// TimeseriesLookupRequest holds request values for submitting to elastic seach
 type TimeseriesLookupRequest struct {
 	Cdid string
 }
 
 var timeseriesTemplate *template.Template
 
+// SetupTimeseries loads templates for use by the timeseries lookup handler and should be done only once
 func SetupTimeseries() error {
 	templates, err := template.ParseFiles("templates/timeseries/lookup.tmpl")
 	timeseriesTemplate = templates
 	return err
 }
 
-func TimeseriesLookupHandler(w http.ResponseWriter, req *http.Request) {
-
-	params := req.URL.Query()
-	reqParams := TimeseriesLookupRequest{Cdid: strings.ToLower(params.Get(":cdid"))}
-
-	var doc bytes.Buffer
-	err := timeseriesTemplate.Execute(&doc, reqParams)
-	if err != nil {
-		log.Debug("Failed to create timeseries query from template", log.Data{"Error": err.Error(), "Params": reqParams})
-		http.Error(w, "Failed to create query", http.StatusInternalServerError)
-		return
+// TimeseriesLookupHandlerFunc returns a http handler function handling search api requests.
+func TimeseriesLookupHandlerFunc(elasticSearchClient ElasticSearcher) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+
+		vars := mux.Vars(req)
```

Comment:
Worth pointing this line out as most of the rest of this change is simply converting a method into a closure., but as the service now uses `mux` rather than `pat`, I needed to change the way it was extracting path variables.